packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -554,12 +555,7 @@ class ComboBoxInternal extends React.Component<IComboBoxInternalProps, IComboBox
       return null;
     }
 
-    const visibleValue = normalizeToString(this._currentVisibleValue);
-    if (comboBox.value !== visibleValue) {
-      return visibleValue;
-    }
-
-    return comboBox.value;
+    return normalizeToString(this._currentVisibleValue);
```

Comment:
This is unrelated to the other changes in the PR, but I came across it while investigating the issue.